code/data_preprocessing/genotype/VCF_QC.ipynb

```diff
@@ -88,23 +93,67 @@
    },
    "source": [
     "## Minimal working example\n",
-    "\n",
+    "The MWE is generated via \n",
+    "```\n",
+    "bcftools query -l get-dosage.ALL.vcf.gz | head -40 > mwe_sample_list\n",
+    "bcftools view -S mwe_sample_list  get-dosage.ALL.vcf.gz > sample_filtered.vcf &\n",
+    "bgzip -c sample_filtered.vcf >  sample_filtered.vcf.gz\n",
+    "tabix -p vcf sample_filtered.vcf.gz\n",
+    "bcftools view --regions chr1 sample_filtered.vcf.gz > chr1_sample_filtered.vcf &\n",
+    "cat chr1_sample_filtered.vcf | head -20000 > mwe_genotype.vcf\n",
+    "```\n",
+    "and was stored here: https://drive.google.com/file/d/1sxxPdPIyKma0mAl8TKwhgyRHlOh0Oyrc/view?usp=sharing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "691cade1-0de8-4c66-a2ef-0c88ccff05af",
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "outputs": [],
+   "source": [
+    "The MWE was used as followed:\n",
     "\n",
     "```\n",
     "sos run VCF_QC.ipynb dbsnp_annotate \\\n",
     "    --genoFile reference_data/00-All.vcf.gz \\\n",
-    "    --cwd reference_data\n",
+    "    --cwd reference_data --container ./bioinfo.sif\n",
     "```\n",
     "\n",
     "```\n",
-    "sos run VCF_QC.ipynb qc \\\n",
-    "    --genoFile data/mwe_genotype.vcf \\\n",
-    "    --dbsnp-variants reference_data/00-All.renamechrs.add_chr.variants.gz \\\n",
-    "    --reference-genome reference_data/GRCh38_full_analysis_set_plus_decoy_hla.fa\n",
-    "    --cwd output/genotype\n",
-    "```\n"
+    "sos run VCF_QC.ipynb qc    \\\n",
+    "--genoFile data/mwe/mwe_genotype.vcf     \\\n",
+    "--dbsnp-variants data/reference_data/00-All.add_chr.variants.gz  \\\n",
+    "--reference-genome data/reference_data/GRCh38_full_analysis_set_plus_decoy_hla.noALT_noHLA_noDecoy_ERCC.fasta   \\\n",
+    "--cwd mwe/output/genotype_1 --container ./bioinfo.sif -J 1 -c csg.yml -q csg  &\n",
+    "```\n",
+    "To produce the following results:"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "d6a2537c-0053-496d-a764-06708b0f2cf6",
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "source": [
+    "- Total TSTV for 19639 known variants before QC: 2.599\n",
```

Comment:
you mean to say "average TS/TV"? You need to show people how to get these numbers and not these numbers themselves (which is obviously not very helpful in this context).

Reply:
This number is the number in the total columns of the tstv output:
```
TS/TV stats:
Sample ,ROS20998065,SM-CJEGP,SM-CJIXD,SM-CJIXI,SM-CTDRJ,SM-CTDT7,SM-CTEM6,SM-CTEMM,ROS20275399,SM-CJGIA,SM-CJIX4,SM-CJIZF,SM-CJK5B,SM-CTDRM,SM-CTEEZ,SM-CTEMY,ROS20251553,SM-CJEIH,SM-CJIXV,SM-CJK3O,SM-CJK45,SM-CTDRN,SM-CTDSR,SM-CTDTV,SM-CJGJ9,SM-CJGNM,SM-CTDST,SM-CTDTW,SM-CTEDO,SM-CTEI5,SM-CJGGV,SM-CJGH9,SM-CJGHJ,SM-CJGHN,SM-CJGII,SM-CJGN1,SM-CJK5A,SM-CTEFK,SM-CJEIJ,SM-CJEJ2,Total
Transitions ,9592,9399,10079,10030,9810,10210,9735,9429,9586,9553,10229,10166,10444,9839,10069,9763,10154,9577,9983,9413,9406,9688,9771,10107,9112,9178,9680,9515,9838,9835,9762,9233,9122,9852,9459,9663,9953,10187,9844,9216,389481
Transversions ,3586,3578,3903,3863,3741,3944,3785,3600,3611,3706,3988,3913,3993,3755,3816,3789,3927,3722,3827,3643,3600,3660,3858,3862,3460,3506,3697,3638,3778,3808,3749,3538,3551,3807,3647,3809,3887,3967,3683,3666,149861
Ts/Tv ,2.675,2.627,2.582,2.596,2.622,2.589,2.572,2.619,2.655,2.578,2.565,2.598,2.616,2.620,2.639,2.577,2.586,2.573,2.609,2.584,2.613,2.647,2.533,2.617,2.634,2.618,2.618,2.615,2.604,2.583,2.604,2.610,2.569,2.588,2.594,2.537,2.561,2.568,2.673,2.514,2.599
```
which is  389481/149861
By how to get these number, do you mean the code to extract this number from the tstv file?

Reply:
This is the last value on the last row of the file, 2.599. You can show users how to check it, in your MWE documentation section: 

```
tail -1 file.txt  | awk -F ","  '{print $NF}'
```

Reply:
It is actually not but I was trying to do sth similar

Reply:
> It is actually not

what do you mean? you shoudl be able to use that number as is. Can you elaborate?

>  I read online saying that awk $NF is limited to 99 columns though,

Good call ... please try another approach

Reply:
The whole file is something like 
```
TS/TV stats:
Sample ,ROS20998065,SM-CJEGP,SM-CJIXD,SM-CJIXI,SM-CTDRJ,SM-CTDT7,SM-CTEM6,SM-CTEMM,ROS20275399,SM-CJGIA,SM-CJIX4,SM-CJIZF,SM-CJK5B,SM-CTDRM,SM-CTEEZ,SM-CTEMY,ROS20251553,SM-CJEIH,SM-CJIXV,SM-CJK3O,SM-CJK45,SM-CTDRN,SM-CTDSR,SM-CTDTV,SM-CJGJ9,SM-CJGNM,SM-CTDST,SM-CTDTW,SM-CTEDO,SM-CTEI5,SM-CJGGV,SM-CJGH9,SM-CJGHJ,SM-CJGHN,SM-CJGII,SM-CJGN1,SM-CJK5A,SM-CTEFK,SM-CJEIJ,SM-CJEJ2,Total
Transitions ,9581,9380,10065,10014,9799,10193,9720,9409,9575,9523,10217,10148,10428,9828,10054,9754,10138,9564,9971,9400,9395,9672,9758,10092,9100,9164,9667,9500,9823,9821,9748,9224,9100,9837,9442,9646,9940,10176,9830,9190,388886
Transversions ,3583,3573,3897,3855,3736,3935,3776,3593,3604,3688,3978,3908,3985,3750,3809,3785,3919,3716,3821,3636,3594,3655,3853,3856,3454,3497,3688,3631,3772,3798,3744,3529,3540,3800,3639,3799,3882,3960,3677,3658,149573
Ts/Tv ,2.674,2.625,2.583,2.598,2.623,2.590,2.574,2.619,2.657,2.582,2.568,2.597,2.617,2.621,2.640,2.577,2.587,2.574,2.610,2.585,2.614,2.646,2.533,2.617,2.635,2.621,2.621,2.616,2.604,2.586,2.604,2.614,2.571,2.589,2.595,2.539,2.561,2.570,2.673,2.512,2.600


Hom/Het stats:
Sample ,ROS20998065,SM-CJEGP,SM-CJIXD,SM-CJIXI,SM-CTDRJ,SM-CTDT7,SM-CTEM6,SM-CTEMM,ROS20275399,SM-CJGIA,SM-CJIX4,SM-CJIZF,SM-CJK5B,SM-CTDRM,SM-CTEEZ,SM-CTEMY,ROS20251553,SM-CJEIH,SM-CJIXV,SM-CJK3O,SM-CJK45,SM-CTDRN,SM-CTDSR,SM-CTDTV,SM-CJGJ9,SM-CJGNM,SM-CTDST,SM-CTDTW,SM-CTEDO,SM-CTEI5,SM-CJGGV,SM-CJGH9,SM-CJGHJ,SM-CJGHN,SM-CJGII,SM-CJGN1,SM-CJK5A,SM-CTEFK,SM-CJEIJ,SM-CJEJ2,Total
Homozygous reference,9925,9913,9789,10181,9823,9350,9699,9831,9832,9436,9046,8760,8942,9615,9222,9612,9356,9751,9267,10096,10419,9763,9326,9354,10273,9970,9496,10054,9837,9452,9775,10130,10196,9678,9986,9674,9331,9374,9500,9750,386784
One ALT,6128,6353,5604,4899,5957,6310,6240,6462,6287,7009,6845,7546,6839,6326,6819,6373,6367,6350,6810,5910,5309,6291,6873,6486,6042,6531,6783,5891,5871,6573,6098,6125,6092,6139,6081,6333,6656,6238,6625,6718,253189
Two ALTs,3518,3300,4179,4485,3789,3909,3628,3270,3446,3101,3675,3255,3787,3626,3522,3583,3845,3465,3491,3563,3840,3518,3369,3731,3256,3065,3286,3620,3862,3523,3697,3314,3274,3749,3500,3556,3583,3949,3441,3065,142635
Missing,2,7,1,8,4,4,6,10,8,27,7,12,5,6,10,5,5,7,5,4,5,1,5,2,2,7,8,8,3,25,3,4,11,7,6,10,3,12,7,40,312


Variant type stats:
Sample ,ROS20998065,SM-CJEGP,SM-CJIXD,SM-CJIXI,SM-CTDRJ,SM-CTDT7,SM-CTEM6,SM-CTEMM,ROS20275399,SM-CJGIA,SM-CJIX4,SM-CJIZF,SM-CJK5B,SM-CTDRM,SM-CTEEZ,SM-CTEMY,ROS20251553,SM-CJEIH,SM-CJIXV,SM-CJK3O,SM-CJK45,SM-CTDRN,SM-CTDSR,SM-CTDTV,SM-CJGJ9,SM-CJGNM,SM-CTDST,SM-CTDTW,SM-CTEDO,SM-CTEI5,SM-CJGGV,SM-CJGH9,SM-CJGHJ,SM-CJGHN,SM-CJGII,SM-CJGN1,SM-CJK5A,SM-CTEFK,SM-CJEIJ,SM-CJEJ2,Total
SNP,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
MNP,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
INS,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
DEL,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
MIXED,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
Multiallelic,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0


Allele count stats:
Sample ,ROS20998065,SM-CJEGP,SM-CJIXD,SM-CJIXI,SM-CTDRJ,SM-CTDT7,SM-CTEM6,SM-CTEMM,ROS20275399,SM-CJGIA,SM-CJIX4,SM-CJIZF,SM-CJK5B,SM-CTDRM,SM-CTEEZ,SM-CTEMY,ROS20251553,SM-CJEIH,SM-CJIXV,SM-CJK3O,SM-CJK45,SM-CTDRN,SM-CTDSR,SM-CTDTV,SM-CJGJ9,SM-CJGNM,SM-CTDST,SM-CTDTW,SM-CTEDO,SM-CTEI5,SM-CJGGV,SM-CJGH9,SM-CJGHJ,SM-CJGHN,SM-CJGII,SM-CJGN1,SM-CJK5A,SM-CTEFK,SM-CJEIJ,SM-CJEJ2,Total
MAC=0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
MAC=1,1,1,2,4,9,1,2,2,5,2,1,1,1,3,3,7,1,1,1,3,2,2,2,1,3,1,1,2,2,2,4,3,1,2,3,1,1,2,2,3,91
MAC=2,15,15,20,14,61,13,33,16,20,20,15,19,17,17,21,21,16,20,18,21,39,18,20,25,19,16,12,15,13,18,19,15,17,61,19,18,15,13,15,18,817
MAC=3,91,100,90,85,86,102,92,101,84,110,99,97,99,89,87,103,91,80,109,94,94,81,119,99,100,86,78,105,113,139,105,83,85,86,88,88,92,106,93,97,3826
MAC=4,90,162,49,85,50,76,108,84,67,149,103,84,123,90,91,83,84,134,77,69,93,76,58,85,99,97,71,123,89,62,117,67,63,60,79,77,76,148,62,92,3552
MAC=5,108,130,90,65,125,100,83,93,112,129,124,146,146,117,87,107,80,96,114,114,87,84,105,89,86,74,121,179,105,89,90,117,104,73,114,120,158,167,86,132,4346
MAC=6,153,187,150,152,172,196,141,165,219,186,232,270,170,167,257,147,169,150,249,188,123,182,185,144,179,166,177,219,144,156,170,211,152,187,165,265,230,194,156,179,7304
MAC=7,282,310,249,321,246,332,324,326,369,313,322,335,309,317,343,265,256,239,311,310,272,287,286,318,267,292,282,318,273,266,355,280,301,307,258,325,321,326,280,259,11952
MAC=8,268,204,228,225,222,260,429,283,233,253,317,435,227,338,325,199,206,224,324,225,261,233,270,341,338,251,362,255,350,241,217,214,251,278,217,332,388,304,333,244,11105
MAC=9,243,238,277,256,305,306,303,242,274,252,354,355,272,333,335,248,268,248,284,226,241,280,294,419,278,243,286,277,267,306,299,260,309,295,343,311,322,286,295,348,11578
MAC=10,239,313,243,272,232,322,254,271,293,308,349,343,280,236,311,256,296,260,321,314,280,272,313,298,369,244,325,358,251,344,323,278,314,277,312,321,352,310,329,255,11838
MAC=11,174,243,206,189,188,244,197,219,268,256,292,245,309,202,255,223,211,289,235,242,215,224,275,207,242,199,262,250,256,233,226,200,245,256,190,278,265,323,272,226,9531
MAC=12,233,232,225,239,216,214,234,217,222,232,342,273,211,197,285,284,225,227,200,268,211,258,211,245,195,219,249,202,202,254,284,209,181,196,203,283,341,265,225,202,9411
MAC=13,189,232,239,249,212,240,266,259,271,246,319,258,228,226,270,240,229,226,227,203,171,263,235,317,162,204,266,246,198,251,248,173,202,226,176,307,237,237,206,218,9372
MAC=14,296,266,298,297,290,232,262,291,250,260,326,248,331,251,297,267,312,271,243,201,213,320,279,285,224,249,290,234,267,258,255,192,218,262,231,211,274,291,275,244,10561
MAC=15,365,287,427,313,335,293,291,313,301,372,434,391,401,359,283,365,426,379,352,270,327,334,383,381,302,357,405,334,309,315,344,288,265,356,302,299,362,328,357,341,13646
MAC=16,368,319,411,379,422,332,312,358,471,426,462,464,449,261,311,513,336,485,455,341,317,344,390,366,299,374,400,320,384,387,441,337,370,352,393,364,349,483,383,294,15222
MAC=17,281,296,352,286,311,332,277,331,256,299,358,306,336,274,318,323,321,247,285,294,293,270,305,303,284,262,292,272,333,254,286,281,319,293,283,304,293,311,339,314,11974
MAC=18,288,360,411,374,379,343,339,320,332,336,391,339,342,352,351,325,348,380,377,360,304,380,308,311,339,376,334,376,343,349,334,351,294,357,356,338,326,337,355,322,13837
MAC=19,330,339,329,292,294,362,280,273,293,312,334,268,300,257,311,299,370,258,315,322,298,309,244,314,267,322,283,243,302,308,285,294,348,312,263,264,270,294,283,283,11924
MAC=20,176,189,180,146,186,224,182,167,172,173,181,170,193,206,192,171,191,172,174,170,173,152,176,192,154,191,183,135,164,166,201,182,167,194,180,181,189,165,195,181,7136
MAC=21,262,304,246,218,285,323,286,252,234,250,282,289,304,276,293,279,339,192,297,216,243,287,319,282,258,252,238,200,254,275,339,262,239,236,305,299,222,245,315,277,10774
MAC=22,242,267,274,164,287,279,215,228,222,178,232,261,288,265,268,277,285,212,230,231,219,215,232,237,242,272,290,210,250,236,301,252,182,227,246,238,251,245,214,280,9744
MAC=23,220,229,237,260,258,312,247,223,256,264,232,300,271,268,277,276,284,219,246,227,218,235,265,299,229,275,223,206,258,231,230,249,231,245,286,253,262,243,256,301,10101
MAC=24,210,222,214,238,235,235,253,246,225,258,221,228,218,228,273,265,264,230,239,216,248,276,224,232,215,206,255,189,214,224,246,213,249,231,252,202,264,242,247,241,9388
MAC=25,242,239,209,248,226,203,240,205,234,223,244,234,214,249,239,262,204,225,251,206,266,249,270,263,233,202,246,215,203,229,225,260,225,268,234,217,240,217,221,209,9289
MAC=26,211,223,195,215,197,214,197,240,249,235,225,234,243,233,231,229,197,223,247,174,206,235,205,239,185,205,231,188,194,209,200,217,231,231,201,187,215,216,209,213,8629
MAC=27,230,174,235,251,165,248,206,263,189,161,197,224,239,244,202,163,221,253,235,159,176,206,224,239,171,214,191,209,231,274,227,190,179,218,189,193,185,191,228,235,8429
MAC=28,242,239,278,292,255,332,275,314,262,291,270,260,299,280,261,284,290,286,286,227,239,276,321,256,255,224,224,251,272,307,280,249,288,297,271,264,272,302,278,301,10950
MAC=29,369,359,326,350,383,370,400,371,313,372,312,367,392,370,377,396,370,306,422,336,348,382,383,403,369,274,347,363,381,302,370,386,308,412,380,345,402,357,366,336,14475
MAC=30,283,277,251,266,261,274,277,247,249,275,284,299,286,290,246,267,264,280,283,278,224,252,300,252,270,258,239,266,283,287,281,270,230,280,233,231,296,283,287,270,10729
MAC=31,311,283,325,307,327,355,316,303,276,345,317,362,342,304,268,300,334,323,316,324,265,305,356,286,310,316,335,318,327,319,276,304,306,282,291,300,333,295,341,290,12493
MAC=32,371,345,354,396,329,388,320,363,330,344,345,415,405,422,390,359,407,398,399,356,322,350,421,357,327,362,354,334,393,350,377,348,307,369,391,320,353,353,415,332,14571
MAC=33,359,307,353,331,338,347,325,341,363,367,302,348,403,349,347,348,377,354,333,393,347,338,353,369,324,383,335,333,304,364,284,388,342,317,318,346,339,334,324,257,13684
MAC=34,369,301,321,306,334,358,351,387,309,370,309,365,389,358,320,304,372,358,385,334,346,360,404,332,317,340,375,399,300,388,323,315,354,323,341,330,320,304,335,368,13774
MAC=35,336,274,278,237,289,290,317,286,324,313,259,334,324,330,334,331,301,349,298,322,319,279,275,316,287,313,350,309,274,335,239,322,278,263,273,306,325,299,321,338,12147
MAC=36,243,215,276,205,257,276,269,242,236,248,217,269,266,256,265,239,266,243,247,236,225,249,245,219,214,223,264,205,273,274,243,248,262,223,243,222,259,228,281,238,9809
MAC=37,318,323,286,284,297,245,310,325,308,326,238,290,312,304,284,274,274,325,301,274,292,237,336,301,222,309,299,243,315,340,230,283,283,293,266,237,241,312,235,306,11478
MAC=38,300,306,291,238,315,301,298,208,288,258,313,280,317,266,321,267,328,246,260,330,245,304,257,239,293,340,209,248,257,332,214,293,260,313,308,326,253,294,309,336,11361
MAC=39,230,243,246,238,262,237,249,249,275,285,261,290,265,256,297,277,275,285,247,277,281,297,273,243,267,290,270,275,263,306,212,250,293,305,266,274,245,236,237,298,10625

```

Reply:
I see. we should then rename the files into : 

````
novel_variant_bcftools_sumstats
novel_variant_snpsift_sumstats
known_...
...
```

and use some R or Python code to get the TS/TV, write that into 

```
known_variants_tstv
novel_variants_tstv
``` 

perhaps you want to change it into snpeff_su

Reply:
> and use some R or Python code to get the TS/TV, write that into

by this, I mean that we should add an R or Python chunk of code below the bash chunk in the summary stats step

Reply:
If it is just for the convienient of extracting the ts/tv, i think my current implementation is good enough?
`grep Ts/Tv {prefix of genoFile}*_variant.snipsift_tstv | rev | cut -d',' -f1 | rev`

Reply:
Agreed. But this is the MWE, we should make sure the command works just copy/paste. 

Please change the code to using the actual file and run it inside a `bash` cell to show people the actual output.

Reply:
done